app/bundles/UserBundle/Security/Authentication/Token/PluginToken.php

```diff
@@ -85,18 +85,22 @@ public function getResponse()
 
     /**
      * {@inheritdoc}
+     *
+     * @return array<mixed>
      */
-    public function serialize()
+    public function __serialize(): array
     {
-        return serialize([$this->authenticatingService, $this->credentials, $this->providerKey, parent::serialize()]);
+        return [$this->authenticatingService, $this->credentials, $this->providerKey, parent::__serialize()];
     }
 
     /**
      * {@inheritdoc}
+     *
+     * @param array<mixed> $data
      */
-    public function unserialize($serialized)
+    public function __unserialize(array $data): void
     {
-        list($this->authenticatingService, $this->credentials, $this->providerKey, $parentStr) = unserialize($serialized);
-        parent::unserialize($parentStr);
+        [$this->authenticatingService, $this->credentials, $this->providerKey, $parentStr] = $data;
```

Comment:
There is a difference between the original code and the new one. Why did you remove the unserialize function? Shouldn't it be like this?:
```suggestion
        [$this->authenticatingService, $this->credentials, $this->providerKey, $parentStr] = unserialize($data);
```

Reply:
I have made changes according to https://github.com/symfony/security/blob/0a70d429b785affd37c48bb39ddfd38911663a57/Core/Authentication/Token/AbstractToken.php#L191